### PR TITLE
text crdt example: RGA

### PR DESCRIPTION
**Pull request by jhellerstein**
An example of a "text CRDT" implemented in Datalog (translated naively to Hydroflow), in Datalog-with-aggregation (translated naively to Hydroflow), and in more aggressive Hydroflow.

Original Datalog versions are in comments. In future would be nice to add as examples for our Datalog parser.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,2 +1,3 @@
 /target*
 /rust.git
+/*.dot
```
Comment (MingweiSamuel) [inline .gitignore:3]: Is this needed? Whats writing dot files to project root?

File: hydroflow/examples/rga/adjacency.rs [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+// children = join::<'tick, 'static>() -> map(|(_n, (_, kids))| kids);
+// visit -> map(|n| (n, ())) -> [0]children;
+// adjacency -> [1]children;
+// children -> stack_buf;
+
+// labels = insertAfter -> map(|(c, _p): (Token, Timestamp)| (c.id, c.value));
+// visit_labels = join::<'tick, 'static>() -> map(|(_id, ((), label))| label);
+// visit -> map(|n| (n, ())) -> [0]visit_labels;
+// labels -> [1]visit_labels;
+// visit_labels -> for_each(|e| print!("{}", e));
```
Comment (MingweiSamuel) [inline hydroflow/examples/rga/adjacency.rs:96]: commented out code?

**Review by MingweiSamuel — APPROVED** [review]

File: .gitignore (hunk 1 above)
Comment (jhellerstein) [reply .gitignore:3]: Im emitting them from the rga example to cwd, which is typically the root of the repo.

Comment (jhellerstein) [reply .gitignore:3]: It's also handy to pipe stdout to a dot file and then visualize the result; again the dot file lands at the top level. I suppose we could ignore dot files at every level..

File: hydroflow/examples/rga/adjacency.rs (hunk 2 above)
Comment (jhellerstein) [reply hydroflow/examples/rga/adjacency.rs:96]: OK I'll keep this somewhere else.

Comment (MingweiSamuel) [reply hydroflow/examples/rga/adjacency.rs:96]: Oh its fine to keep here, just not sure what it is for

File: .gitignore (hunk 1 above)
Comment (MingweiSamuel) [reply .gitignore:3]: Ok sounds good to me to keep as-is

File: hydroflow/examples/rga/adjacency.rs (hunk 2 above)
Comment (jhellerstein) [reply hydroflow/examples/rga/adjacency.rs]: it's an implementation of a stack that pops one item per tick. I'll save it on the side -- something we should discuss.

**Pull request merged**